### PR TITLE
add pify.all module support

### LINKED ISSUE #1: Add `pify.all()` method to promisify all methods in an object
Example:

```js
var pify = require('pify');
var fs = pify(require('fs'));

fs.readFile('unicorn.json').then(function () {});
```

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #1 

Not sure if this is what you had in mind, but I think it works as intended!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,3 +29,17 @@ module.exports = function (fn, P) {
 		});
 	};
 };
+
+pify.all = function (module, P) {
+	var pifyModule = {};
+
+	for (var method in module) {
+		if (isFn(module[method])) {
```
Comment (sindresorhus) [inline index.js]: Maybe also make sure it's not an inherited function?

`if (isFn(module[method]) && ({}).hasOwnProperty.call(module, method)) {`

File: index.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
 'use strict';
-module.exports = function (fn, P) {
+var isFn = function (fn) {
+	return ({}).toString.call(fn) === '[object Function]';
+};
+
+var pify = function (fn, P) {
```
Comment (sindresorhus) [inline index.js]: `var pify = module.exports = function (fn, P) {`

File: index.js [hunk 3]
```diff
@@ -25,3 +29,17 @@ module.exports = function (fn, P) {
 		});
 	};
 };
+
+pify.all = function (module, P) {
+	var pifyModule = {};
```
Comment (sindresorhus) [inline index.js]: `var ret = {};`

File: readme.md [hunk 4]
```diff
@@ -20,6 +20,14 @@ pify(fs.readFile)('package.json', 'utf8').then(function (data) {
 	console.log(JSON.parse(data).name);
 	//=> 'pify'
 });
+
+// Wrap modules entirely.
```
Comment (sindresorhus) [inline readme.md]: `// Wrap entire modules`

File: readme.md [hunk 5]
```diff
@@ -20,6 +20,14 @@ pify(fs.readFile)('package.json', 'utf8').then(function (data) {
...
+var promiseFs = pify.all(fs);
+
+promiseFs.readFile('package.json', 'utf-8').then(function (data) {
```
Comment (sindresorhus) [inline readme.md]: `utf-8` => `utf8

**Comment by sindresorhus** [comment]
Nice! Thanks for working on this :)

File: index.js [hunk 6]
```diff
@@ -25,3 +29,17 @@ module.exports = function (fn, P) {
...
+
+	for (var method in module) {
+		if (isFn(module[method])) {
+			pifyModule[method] = pify(module[method], P);
```
Comment (sindresorhus) [inline index.js]: Actually, you still need to copy over non-methods, just not promisify them.

```js
var x = module[method];
pifyModule[method] = isFn(x) ? pify(x, P) : x;
```

**Comment by stephenplusplus** [comment]
Sorry for the sloppiness! Updated.

**Pull request closed without merging**